src/hotspot/share/memory/heapShared.cpp

```diff
@@ -1278,33 +1267,32 @@ void HeapShared::init_subgraph_entry_fields(ArchivableStaticFieldInfo fields[],
   }
 }
 
-void HeapShared::init_subgraph_entry_fields(Thread* THREAD) {
+void HeapShared::init_subgraph_entry_fields(TRAPS) {
   assert(is_heap_object_archiving_allowed(), "Sanity check");
   _dump_time_subgraph_info_table = new (ResourceObj::C_HEAP, mtClass)DumpTimeKlassSubGraphInfoTable();
   init_subgraph_entry_fields(closed_archive_subgraph_entry_fields,
                              num_closed_archive_subgraph_entry_fields,
-                             THREAD);
+                             THREAD /* aborts on exception */);
```

Comment:
So these should really be CATCH then.

Reply:
This code makes me wonder if Thread* THREAD doesn't make the intent of this code clearer.